### DISCUSSION TITLE
Handle 429 Too Many Requests error from API?

### DISCUSSION THREAD

**Discussion by michaelwittig** [bbc/cloudflare-queue-consumer#23] [Ideas]
Hi,

I recently encountered 429 errors (the default limit seems to be 1200 req/5min). The error is emitted as an `error` event.

```
{"name":"ProviderError","code":"ResponseNotOk","time":"2024-05-16T11:32:32.175Z","status":429,"statusText":"Too Many Requests","url":"https://api.cloudflare.com/client/v4/accounts/***/queues/***/messages/pull","message":"[429 - Too Many Requests] https://api.cloudflare.com/client/v4/accounts/***/queues/*/messages/pull"}
```

I wonder if 429 errors should be handled in `src/lib/cloudflare.ts` by using an exponential backoff strategy (which the AWS SDK implements for us when using SQS).

Please let me know your thoughts.

Michael

**Comment by nicholasgriffintn** [comment]
Sorry, just getting back to this, I think I've implemented something can handle this, it's a really good suggestion.

Just need to get some more stuff out the way on this, happy to get suggestions though if there are improvements to be had :)

**Comment by michaelwittig** [comment]
No worries. I stopped using the pull based approach because of the very low api rate limit (1200 requests per five minutes).
